tests/models/autoencoders/test_models_vq.py

```diff
@@ -98,3 +99,18 @@ def test_output_pretrained(self):
         expected_output_slice = torch.tensor([-0.0153, -0.4044, -0.1880, -0.5161, -0.2418, -0.4072, -0.1612, -0.0633, -0.0143])
         # fmt: on
         self.assertTrue(torch.allclose(output_slice, expected_output_slice, atol=1e-3))
+    def test_loss_pretrained(self):
+        model = VQModel.from_pretrained("fusing/vqgan-dummy")
+        model.to(torch_device).eval()
+
+        torch.manual_seed(0)
+        backend_manual_seed(torch_device, 0)
+
+        image = torch.randn(1, model.config.in_channels, model.config.sample_size, model.config.sample_size)
+        image = image.to(torch_device)
+        with torch.no_grad():
+            output = model(image, return_loss=True).commit_loss.cpu()
+        # fmt: off
+        expected_output = torch.tensor([0.1936])
+        # fmt: on
+        self.assertTrue(torch.allclose(output, expected_output, atol=1e-3))
```

Comment:
I think instead of this test, we could add a smoke test like this:
https://github.com/huggingface/diffusers/blob/main/examples/text_to_image/test_text_to_image.py

Does that make sense?

Reply:
@sayakpaul hi! I just did the tests. Let me know what you think! I did find a part of this script that makes checkpointing steps 1 and 2 exactly the same as we count the generator step as one step and then the discriminator step as 1 step for getting the full loss. Let me know if you think that's ok.
Also, should I delete the vq test? I just left it there since that was testing a new piece of code I introduced

Reply:
I think this test can live too. I don't find any problems.